File: 3115-maximum-prime-difference/3115-maximum-prime-difference.cpp
class Solution {
public:
       bool isPrime(int n) {
        if (n < 2) {
            return false;
        }
        if (n == 2) {
            return true;
        }
        for (int i = 2; i <= sqrt(n); i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }
    int maximumPrimeDifference(vector<int>& nums) {
        for(int i=0;i<nums.size();i++)
        {
            if(isPrime(nums[i])) nums[i]=1; 
            else nums[i]=0; 
        }
     int first=1e9; int last=-1e9; 
      for(int i=0;i<nums.size();i++)
      {
          if(nums[i]==1) {first=min(first,i); last=max(last,i);}
      }
        return abs(first-last);
    }
};

File: 3115-maximum-prime-difference/3115-maximum-prime-difference.java
class Solution {
    public boolean isprime(int n)
    {
        if (n <= 1)
            return false;

        for (int i = 2; i <= Math.sqrt(n); i++)
            if (n % i == 0)
                return false;

        return true;
    }
    public int maximumPrimeDifference(int[] nums) {
        Stack<Integer> stidx=new Stack<>();
        int n=nums.length;
        for(int i=0;i<n;i++)
        {
            if(isprime(nums[i]))
            {
                stidx.push(i);
            }
        } int ans=0;
        for(int j=n-1;j>=0;j--)
        {
            while(!stidx.isEmpty()&&isprime(nums[j]))
            {
                ans=Math.max(ans,j-stidx.pop());
            }
        }
        return ans;
    }
}